builtins-default-costs/src/lib.rs

```diff
@@ -36,77 +34,78 @@ lazy_static! {
     /// https://github.com/solana-labs/solana/issues/29595.
     static ref BUILTIN_INSTRUCTION_COSTS: AHashMap<Pubkey, BuiltinCost> = [
     (
-        solana_stake_program::id(),
+        solana_system_program::id(),
         BuiltinCost {
-            native_cost: solana_stake_program::stake_instruction::DEFAULT_COMPUTE_UNITS,
-            core_bpf_migration_feature: Some(feature_set::migrate_stake_program_to_core_bpf::id()),
+            native_cost: solana_system_program::system_processor::DEFAULT_COMPUTE_UNITS,
+            core_bpf_migration_feature: BUILTINS[0].core_bpf_migration_config.as_ref().map(|c| c.feature_id),
```

Comment:
wdyt to replace `core_bpf_migration_feature: Option<Pubkey>` with something like `builtins_index: usize`?

And if can move `native_cost` to BUILTINS, then `BUILTIN_INSTRUCTION_COSTS` could be `AhashMap<Pubkey, usize /*index of BUILTINS*/>`, that basically provides a "lookup by Pubkey" function.  So it'd look like `(solana_system_program::id(), 0), (solana_vote_program::id(), 1), ...`

Reply:
Yeahhh that's a good idea, probably a lot cleaner. I wonder if instead of a "core bpf migration" crate, we can just make a "builtins" crate, and then I don't have to move as much stuff. Then it also makes sense why we'd overload the list with a bunch of configs. Wdyt?

Reply:
`builtins` crate with its static array of builtins would be great. It will be the single source of truth. With that, might be able to remove `builtins-default-costs` crate too - its remaining function is Ahash lookup, and (copy and) host "native cost" - this is soon to be obsolete.

Reply:
Alright, then that's what I'll do. Once I throw it up I'll abandon this PR.

Reply:
@tao-stones what do you think of this? https://github.com/anza-xyz/agave/pull/3866

We could combine our crates (ie. `builtin-default-costs` and `builtins`) but there are some large dependencies (program builtin functions) that `builtins` has to pull in, and `builtin-default-costs` does not.